numpy/ma/core.py

```diff
@@ -6206,16 +6227,23 @@ def __str__(self):
         return str(masked_print_option._display)
 
     def __repr__(self):
-        return 'masked'
-
-    def flatten(self):
-        return masked_array([self._data], dtype=float, mask=[True])
+        if self is masked:
+            return 'masked'
+        else:
+            # something is wrong, make it obvious
```

Comment:
Well, that's the hope. There were many different ways that this could happen before, and this change made it easy to spot.

It also means that subclasses of `MaskedConstant` don't masquerade as `masked` either.